src/main/java/org/openapitools/jackson/nullable/JsonNullable.java

```diff
@@ -84,6 +157,141 @@ public void ifPresent(
         }
     }
 
+    /**
+     * If a value is present, performs the given action with the value,
+     * otherwise performs the given empty-based action.
+     *
+     * @param action the action to be performed, if a value is present
+     * @param undefinedAction the empty-based action to be performed, if no 
+     *        value is present
+     * @throws NullPointerException if a value is present and the given action
+     *         is null, or no value is present and the given empty-based action
+     *         is null
+     *
+     * @since 0.2.8
+     */
+    public void ifPresentOrElse( Consumer<? super T> action, Runnable undefinedAction ) {
+        if (this.isPresent) {
+            action.accept(value);
+        }
+        else {
+            undefinedAction.run();
+        }
+    }
+
+    /**
+     * If a value is present, and the value matches the given predicate, returns
+     *  a JsonNullable describing the value, otherwise returns an undefined
+     * JsonNullable.
+     *
+     * @param predicate the predicate to apply to a value, if present
+     * @return a JsonNullable describing the value of this JsonNullable,
+     *   if a value is present and the value matches the given predicate,
+     *   otherwise an undefined JsonNullable
+     * @throws NullPointerException if the predicate is null
```

Comment:
This only happens if the predicate is null AND the value is not present. (Slightly different than the JDK Optional behavior I think.)

Either the throws description should be updated, or the method should be updated.

Reply:
Probably better to update the method and just be consistent with the Optional behavior.

Reply:
Test and fix for this issue is complete.